xbmc/filesystem/Directory.h

```diff
@@ -69,8 +69,8 @@ class CDirectory
                            , CFileItemList &items
                            , const CHints &hints);
 
-  using DirectoryEnumerationCallback = std::function<void(const std::shared_ptr<CFileItem>& item)>;
-  using DirectoryFilter = std::function<bool(const std::shared_ptr<CFileItem>& folder)>;
+  using DirectoryEnumerationCallback = std::function<void(std::shared_ptr<CFileItem>& item)>;
+  using DirectoryFilter = std::function<bool(std::shared_ptr<CFileItem>& folder)>;
```

Comment:
Why is it necessary, or why is beneficial, to remove the `const` here?

Reply:
That was necessary in AddVideoExtras() due to GetArtwork() modifying the parameter item. Adds art, adds art to its actors, manipulates the video tag a bit (though that part doesn't to survive outside of GetArtwork)

I'm not sure EnumerateDirectory() should have to make a const guarantee. The callers are free to use const auto& item.

Reply:
I agree that callers should be able to modify the `CFileItem`. And that's already possible, this PR compiles fine without touching Directory.cpp/h. What the `const` prevents is to change what the `std::shared_ptr` points to, and I believe callers shouldn't be able to change that.

(In hindsight I should have made the filter `std::function<bool(const std::shared_ptr<const CFileItem>& folder)>` as it's a [predicate](https://en.cppreference.com/w/cpp/named_req/Predicate)) .

Reply:
You're right. I had compilation problems earlier that I resolved that way but it wasn't necessary. Leaving the const in.